src/components/common/ViewerSidebar.jsx

```diff
@@ -1225,6 +1226,13 @@ class ViewerSidebar extends Component {
                             {markerTable}
                             {legendTable}
                             {downloadSubLoomButton()}
+                            <GProfilerModal
+                                numFeatures={md.genes.length}
+                                clusteringID={
+                                    activeFeatures[i].metadata.clusteringID
+                                }
+                                clusterID={activeFeatures[i].metadata.clusterID}
+                            />
```

Comment:
There is no check to ensure that we're looking at a clustering, or if there are any marker genes. This currently causes the button to appear on all searches and when there are no marker genes, can crash the frontend.

`md.genes.length` shouldn't be called without confirming that `md.genes` isn't null.

Reply:
Indeed, good catch